scripts/src/java/org/oppia/android/scripts/ci/ComputeChangedFiles.kt

```diff
@@ -0,0 +1,330 @@
+   *     protos should be printed
+   * @param baseCommit see [GitClient]
+   * @param computeAllFilesSetting whether all files should be outputted versus only the ones which
+   *     are changed by local changes in the repository
+   */
+  fun compute(
+    pathToRoot: String,
+    pathToOutputFile: String,
+    baseCommit: String,
+    computeAllFilesSetting: Boolean
+  ) {
+    val rootDirectory = File(pathToRoot).absoluteFile
+    check(rootDirectory.isDirectory) { "Expected '$pathToRoot' to be a directory" }
+    check(rootDirectory.list()?.contains("WORKSPACE") == true) {
+      "Expected script to be run from the workspace's root directory"
+    }
+
+    println("Running from directory root: $rootDirectory.")
+
+    val gitClient = GitClient(rootDirectory, baseCommit, commandExecutor)
+    println("Current branch: ${gitClient.currentBranch}.")
+    println("Most recent common commit: ${gitClient.branchMergeBase}.")
+
+    val currentBranch = gitClient.currentBranch.lowercase(Locale.US)
+    val changedFiles = if (computeAllFilesSetting || currentBranch == "develop") {
+      computeAllFiles(rootDirectory, pathToRoot)
+    } else computeChangedFilesForNonDevelopBranch(gitClient, rootDirectory)
+
+    val ktFiles = changedFiles.filter { it.endsWith(".kt") }
+    val filteredFiles = filterFiles(ktFiles)
+
+    val changedFileBuckets = bucketFiles(filteredFiles)
+    val encodedFileBucketEntries = changedFileBuckets
+      .associateBy { it.toCompressedBase64() }
+      .entries.shuffled()
+
+    File(pathToOutputFile).printWriter().use { writer ->
+      encodedFileBucketEntries.forEachIndexed { index, (encoded, bucket) ->
+        writer.println("${bucket.cacheBucketName}-shard$index;$encoded")
+      }
+    }
+  }
+
+  private fun computeAllFiles(
+    rootDirectory: File,
+    pathToRoot: String
+  ): List<String> {
+    val testFileExemptiontextProto = "scripts/assets/test_file_exemptions"
+    val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptiontextProto)
+      .testFileExemptionList
+      .filter { it.testFileNotRequired }
+      .map { it.exemptedFilePath }
+
+    val searchFiles = RepositoryFile.collectSearchFiles(
+      repoPath = pathToRoot,
+      expectedExtension = ".kt",
+      exemptionsList = testFileExemptionList
+    )
+
+    return searchFiles
+      .filter { it.name.endsWith(".kt") && !it.name.endsWith("Test.kt") }
+      .map { rootDirectory.toPath().relativize(it.toPath()).toString() }
+  }
+
+  private fun computeChangedFilesForNonDevelopBranch(
+    gitClient: GitClient,
+    rootDirectory: File
+  ): List<String> {
+    return gitClient.changedFiles
+      .map { File(rootDirectory, it) }
+      .filter { it.exists() }
+      .map { rootDirectory.toPath().relativize(it.toPath()).toString() }
+  }
+
+  private fun filterFiles(files: List<String>) : List<String> {
+    // Filtering out files that need to be ignored.
```

Comment:
The targets that belonged to the `//instrumentation/src/javatests/org/oppia/android/instrumentation/player` path were skipped in the **[ComputeAffectedTests.kt](https://github.com/oppia/oppia-android/blob/develop/scripts/src/java/org/oppia/android/scripts/ci/ComputeAffectedTests.kt#L129)**, 

```kotlin
private fun filterTargets(testTargets: List<String>): List<String> {
    // Filtering out the targets to be ignored.
    return testTargets.filter { targetPath ->
      !targetPath
        .startsWith(
          "//instrumentation/src/javatests/org/oppia/android/instrumentation/player",
          ignoreCase = true
        )
    }
  }
```

since coverages is also essentially running tests but collecting coverages additionally, thought to just follow along with whatever set up was done with the existing utility.

Reply:
Ah. In that case, I suggest updating the comment like so:

``Filter out instrumentation files since code coverage only runs on unit tests.``

Reply:
Done, updated with the same.